examples/cpp/function-execution/main.cpp

```diff
@@ -38,147 +34,113 @@
+  auto regionFactory = cache.createRegionFactory(RegionShortcut::PROXY);
+  auto regPtr0 = regionFactory.setPoolName("pool").create("partition_region");
 
-// The Execute Function QuickStart example.
-int main(int argc, char** argv) {
-  try {
-    // Create CacheFactory using the settings from the geode.properties file by
-    // default.
-    auto cache = CacheFactory()
-        .set("log-level", "none")
-        .create();
-
-    std::cout << "Created CacheFactory\n";
-
-    auto pool = cache.getPoolManager()
-        .createFactory()
-        .setSubscriptionEnabled(true)
-        .addServer("localhost", 50505)
-        .addServer("localhost", 40404)
-        .create("pool");
-
-    // Create the example Region Programmatically
-    auto regionFactory = cache.createRegionFactory(RegionShortcut::PROXY);
-    auto regPtr0 = regionFactory.setPoolName("pool").create("partition_region");
-
-    std::cout << "Created the Region\n";
-
-    regPtr0->registerAllKeys();
-    char buf[128];
-
-    auto resultList = CacheableVector::create();
-    for (int i = 0; i < EXAMPLE_ITEM_COUNT; i++) {
-      sprintf(buf, "VALUE--%d", i);
-      auto value(CacheableString::create(buf));
-
-      sprintf(buf, "KEY--%d", i);
-      auto key = CacheableKey::create(buf);
-      regPtr0->put(key, value);
-    }
+  return regPtr0;
+}
 
-    auto routingObj = CacheableVector::create();
-    for (int i = 1; i < EXAMPLE_ITEM_COUNT; i+=2) {
-      sprintf(buf, "KEY--%d", i);
-      auto key = CacheableKey::create(buf);
-      routingObj->push_back(key);
-    }
+void populateRegion(const std::shared_ptr<Region>& regionPtr) {
+  for (int i = 0; i < EXAMPLE_ITEM_COUNT; i++) {
+    std::stringstream keyStream;
+    std::stringstream valueStream;
+    keyStream << "KEY--" << i;
+    valueStream << "VALUE--" << i;
+    auto value(CacheableString::create(valueStream.str().c_str()));
 
-    std::cout << "test data independent function with result on one server\n";
-    auto exc = FunctionService::onServer(regPtr0->getRegionService());
-    if(auto executeFunctionResult = exc.withArgs(routingObj).execute(getFuncIName)->getResult()) {
-      for (auto &arrayList: *executeFunctionResult) {
-        for (auto &cachedString: *std::dynamic_pointer_cast<CacheableArrayList>(arrayList)) {
-          resultList->push_back(cachedString);
-        }
-      }
-      sprintf(buf, "get: result count = %lu\n", resultList->size());
-      std::cout << buf;
-      int i = 0;
-      for (auto &cachedString: *resultList) {
-        sprintf(
-            buf, "get result[%d]=%s\n", i,
-            std::dynamic_pointer_cast<CacheableString>(cachedString)->value().c_str());
-        std::cout << buf;
-        ++i;
+    auto key = CacheableKey::create(keyStream.str().c_str());
+    regionPtr->put(key, value);
+  }
+}
+
+std::shared_ptr<CacheableVector> populateQueryObject() {
+  auto routingObj = CacheableVector::create();
+  for (int i = 1; i < EXAMPLE_ITEM_COUNT; i+=2) {
```

Comment:
Why every other value? Why not fixed values here. This stinks like the integration test. This should be a good clean room function example.